### ISSUE TITLE
new rule 'const-parameters'

### ISSUE THREAD

**Issue by ajafff** [ajafff/tslint-consistent-codestyle#34]
Similar to tslint's `no-parameter-reassignment`, but allows users to declare which parameter should be considered const:

```ts
function fn(/**@const*/ foo, bar) {
    foo++; // error
    bar++; // no error
}
```

Recognise `/**@const*/` and `/**@constant*/`

Ref: https://github.com/Microsoft/TypeScript/issues/18497
